### PR TITLE
DOC: fix docstring of timedaltas.ceil

### LINKED ISSUE #59902: docs: update `freq` description for `pd.Timedelta.ceil` to match `DatetimeIndex.ceil`'s `freq` description
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
td = pd.Timedelta('1001ms')
td.ceil("s")
td.ceil("min")
td.ceil("m")
```

### Issue Description

Calling td.ceil("m") causes the error:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "timedeltas.pyx", line 2011, in pandas._libs.tslibs.timedeltas.Timedelta.ceil
  File "timedeltas.pyx", line 1934, in pandas._libs.tslibs.timedeltas.Timedelta._round
  File "timedeltas.pyx", line 2309, in pandas._libs.tslibs.timedeltas.get_unit_for_round
  File "offsets.pyx", line 756, in pandas._libs.tslibs.offsets.BaseOffset.nanos.__get__
ValueError: <MonthEnd> is a non-fixed frequency
```

### Expected Behavior

td.ceil("m") should behave the same as td.ceil("min")

The documentation for [pandas.Timedelta.ceil](https://pandas.pydata.org/docs/reference/api/pandas.Timedelta.ceil.html#pandas.Timedelta.ceil) states that the freq parameter uses the same units as class constructor Timedelta.

The documentation for [pandas.Timedelta](https://pandas.pydata.org/docs/reference/api/pandas.Timedelta.html#pandas.Timedelta) lists (‘minutes’, ‘minute’, ‘min’, or ‘m’) as one of the possible value lines. This suggests to me that "min" and "m" are equivalent.

It seems that either the documentation could be clearer or there is a bug?

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : 0691c5cf90477d3503834d983f69350f250a6ff7
python                : 3.12.4
python-bits           : 64
OS                    : Windows
OS-release            : 11
Version               : 10.0.22631
machine               : AMD64
processor             : Intel64 Family 6 Model 154 Stepping 3, GenuineIntel
byteorder             : little
LC_ALL                : None
LANG                  : en_US.UTF-8
LOCALE                : English_United Kingdom.1252

pandas                : 2.2.3
numpy                 : 2.0.1
pytz                  : 2024.2
dateutil              : 2.9.0.post0
pip                   : 24.1
Cython                : None
sphinx                : None
IPython               : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : None
blosc                 : None
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : None
html5lib              : None
hypothesis            : None
gcsfs                 : None
jinja2                : 3.1.4
lxml.etree            : None
matplotlib            : None
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : None
pandas_gbq            : None
psycopg2              : None
pymysql               : None
pyarrow               : None
pyreadstat            : None
pytest                : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : 1.14.0
sqlalchemy            : None
tables                : None
tabulate              : None
xarray                : None
xlrd                  : None
xlsxwriter            : None
zstandard             : None
tzdata                : 2024.2
qtpy                  : None
pyqt5                 : None

</details>

### PR DESCRIPTION
**Pull request by eightyseven**
- [ ] closes #59902
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by eightyseven** [comment]
I check the function `timedelta.ceil()` and find that only `D`,`H`,`min`,`s`,`ms`,`us`,`ns` are valid inputs but maybe more like `m` will be added later?so I didn't list all the possilble value in my PR

**Comment by eightyseven** [comment]
I check the function `timedelta.ceil()` and find that only D,H,min,s,ms,us,ns are valid inputs but maybe more like m will be added later?so I didn't list all the possilble value in my PR

**Comment by eightyseven** [comment]
I check the function timedelta.ceil() and find that only D,H,min,s,ms,us,ns are valid inputs but maybe more like 'm' will be added later?so I didn't list all the possilble value in the docstring

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -2176,8 +2176,10 @@ class Timedelta(_Timedelta):
         Parameters
         ----------
         freq : str
-            Frequency string indicating the ceiling resolution.
-            It uses the same units as class constructor :class:`~pandas.Timedelta`.
+            The frequency level to ceil the index to. Must be a fixed
```
Comment (mroeschke) [inline pandas/_libs/tslibs/timedeltas.pyx:2179]:
```suggestion
            Frequency string indicating the ceiling resolution. Must be a fixed
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @eightyseven

**Pull request merged**